changelogs-sdk.gradle

```diff
@@ -0,0 +1,406 @@
+				}
+				else {
+					matcher = gitLogLine =~ /^\+\s*.+ group: .*, name: \"(.+)", .*version: \"(.*)\"/
+
+					if (!matcher.find()) {
+						matcher = gitLogLine =~ /^\+([^ \t]+)=([0-9.]+)/
+
+						if (!matcher.find()) {
+							return
+						}
+					}
+
+					String dependencyName = matcher.group(1)
+					String dependencyVersion = matcher.group(2)
+
+					dependencies = dependencies + "\n" + _format("- [" + gitLogAttributes["TICKET-NUMBER"] + "]: Update the " + dependencyName + " dependency to version " + dependencyVersion + ".")
+
+					ticketNumbers.add(gitLogAttributes["TICKET-NUMBER"])
+				}
+			}
+
+			String descriptions = ""
+
+			if (changelogDescriptionsFile.exists()) {
+
+				String escapedDotVersion = version.replace(".", "\\.")
+
+				matcher = changelogDescriptionsFile.text =~ /(?s)${escapedDotVersion}:description(.+?)(?:(?=\d+\.\d+\.\d+:description)|$)/
+
+				if (matcher.find()) {
+					descriptions = "\n" + matcher.group(1).trim()
+				}
+			}
+
+			String endRange = commitRange.split("\\.\\.")[1]
+
+			standardOutputByteArrayOutputStream = new ByteArrayOutputStream()
+
+			exec {
+				commandLine "git", "log", "-1", "--pretty=format:%H:%cd:%s", "--date=short", "${endRange}"
+				standardOutput = standardOutputByteArrayOutputStream
+			}
+
+			String publishDate = standardOutputByteArrayOutputStream.toString().split(":")[1]
+
+			if ((commits.length() > 0) || (dependencies.length() > 0) || (descriptions.length() > 0)) {
+				text += """\
+						|
+						|
+						|## ${version} - ${publishDate}""".stripMargin()
+			}
+
+			if (commits.length() > 0) {
+				text += """\
+						|
+						|
+						|### Commits""".stripMargin()
+
+				text += commits
+			}
+
+			if (dependencies.length() > 0) {
+				text += """\
+						|
+						|
+						|### Dependencies""".stripMargin()
+
+				text += dependencies
+			}
+
+			if (descriptions.length() > 0) {
+				text += """\
+						|
+						|
+						|### Description""".stripMargin()
+
+				text += descriptions
+			}
+		}
+
+		if (ticketNumbers.size > 0) {
+			text += """\
+					|
+					|
+					|## Markdown Links""".stripMargin()
+
+			Matcher matcher = ticketNumbers =~ /([A-Z]+-\d+)/
+
+			List<String> tickets = []
+
+			while (matcher.find()) {
+				tickets.add(matcher.group(1))
+			}
+
+			tickets = tickets.flatten().unique().sort{
+				ticket1, ticket2 ->
+
+				String ticketType1 = ticket1.substring(0, ticket1.indexOf("-"))
+				String ticketType2 = ticket2.substring(0, ticket2.indexOf("-"))
+
+				String ticketNumber1 = ticket1.substring(ticket1.indexOf("-") + 1)
+				String ticketNumber2 = ticket2.substring(ticket2.indexOf("-") + 1)
+
+				if (ticketType1 == ticketType2) {
+					return Integer.valueOf(ticketNumber1) - Integer.valueOf(ticketNumber2)
+				}
+
+				return ticketType1.compareTo(ticketType2)
+			}
+
+			tickets.each {
+				String ticket ->
+
+				text += "\n[" + ticket + "](https://issues.liferay.com/browse/" + ticket + ")"
```

Comment:
I'm not sure why the links don't work, but this is the markdown syntax that works for me. 

`text += "\n[" + ticket + "]: https://issues.liferay.com/browse/" + ticket `

Reply:
I think that is the definition of the link that can be referred by other links.
like:
```
[1]: http://google.com/        "Google"
[2]: http://search.yahoo.com/  "Yahoo Search"
[3]: http://search.msn.com/    "MSN Search"

[Link] [2]
```